components/concept/engine/src/main/java/mozilla/components/concept/engine/webextension/WebExtension.kt

```diff
@@ -81,6 +82,29 @@ abstract class WebExtension(
      * null if port is from a background script.
      */
     abstract fun disconnectPort(name: String, session: EngineSession? = null)
+
+    /**
+     * Value type that represents the state of a browser action within a [WebExtension].
+     *
+     * @property title The title of the web extension to be visible in the user interface.
+     * @property enabled Indicate if the browser action should be enabled or disabled.
+     * @property icon The image for this browser icon.
+     * @property uri The url to get the HTML document representing the internal user interface of the extension.
+     * @property badgeText The browser action's badge text.
+     * @property badgeTextColor The browser action's badge text color.
+     * @property badgeBackgroundColor The browser action's badge background color.
+     * @property onClick A callback to be executed when this browser action is clicked.
+     */
+    data class BrowserAction(
```

Comment:
Let's move this out to a top-level type again. If you're worried about the name clash we can use `import as`.